Update styling for Create User Profile desktop

T-Allen opened the pull request:
### Quick Description of the PR

Created a CreateUserProfile page and added styling for desktop and mobile.

### An Overview of the Changes
- New page CreateUserProfile.js
- Added 'create-profile' route to routes.js
- Added new image asset (.svg) from Figma to 'create-profile-images' folder in 'assets' folder.
- Installed 'typeface-dm-sans' package in client to display DM Sans font. 

Resolves #176

anniegiang commented on client/src/pages/CreateUserProfile.js:
```diff
@@ -0,0 +1,219 @@
+            <CheckboxGroup description="I am traveling" />
+            <CheckboxGroup description="Don't show my address" />
+            <InputGroup>
+                <Title>I want to</Title>
```
Need to update the styling a bit:

current:
![Screen Shot 2020-05-07 at 2 36 10 PM](https://user-images.githubusercontent.com/42228386/81347302-b2e04300-9070-11ea-96c5-b61a5d8eb205.png)

figma:
![Screen Shot 2020-05-07 at 2 36 19 PM](https://user-images.githubusercontent.com/42228386/81347312-b83d8d80-9070-11ea-81b9-19a81a767708.png)
![Screen Shot 2020-05-07 at 2 36 27 PM](https://user-images.githubusercontent.com/42228386/81347318-bb387e00-9070-11ea-8ee4-fcea60b8a533.png)

anniegiang commented on client/src/pages/CreateUserProfile.js:
```diff
@@ -0,0 +1,219 @@
+                    </div>
+                </DropdownMenu>
+            </InputGroup>
+            <CheckboxGroup description="I am traveling" />
```
Need to update styling abit to match designs:

current:
![Screen Shot 2020-05-07 at 2 36 51 PM](https://user-images.githubusercontent.com/42228386/81347397-e1f6b480-9070-11ea-9859-10a4054f4d55.png)

figma:
![Screen Shot 2020-05-07 at 2 36 47 PM](https://user-images.githubusercontent.com/42228386/81347409-ea4eef80-9070-11ea-996a-ece4ce54763d.png)
![Screen Shot 2020-05-07 at 2 42 34 PM](https://user-images.githubusercontent.com/42228386/81347446-018ddd00-9071-11ea-88da-b529583621d1.png)

anniegiang replied:
For the checkboxes, need to update styling to match Figma

current:
![Screen Shot 2020-05-07 at 2 35 30 PM](https://user-images.githubusercontent.com/42228386/81347488-1d917e80-9071-11ea-9e9a-3ec7aec29d24.png)

figma:
![Screen Shot 2020-05-07 at 2 35 34 PM](https://user-images.githubusercontent.com/42228386/81347502-24b88c80-9071-11ea-9906-27a62ca918c3.png)

anniegiang commented on client/src/pages/CreateUserProfile.js:
```diff
@@ -0,0 +1,219 @@
+            <InputGroup>
+                <SubmitButton primary="true">Create Profile</SubmitButton>
+                <CheckboxGroup>
+                    <UnderlineLink>By signing up, I agree to the <a href="#">Terms and Conditions</a>
```
The order of the UnderlineLinks are switched, should be Privacy and Policy first. 

You can find the `href` values in `routes.js` for both urls to both pages.

anniegiang commented on client/src/pages/CreateUserProfile.js:
```diff
@@ -0,0 +1,219 @@
+    }
+`;
+
+const ProfileFormGroup = styled.form`
```
The background color of the form section looks different than on Figma, I think the color it should be Selago.

anniegiang commented on client/src/pages/CreateUserProfile.js:
```diff
@@ -0,0 +1,219 @@
+import PersonalDataImage from "../assets/create-profile-images/personal_data.svg";
+
+require('typeface-dm-sans');
+require('typeface-work-sans');
```
For some reason I'm unable to `npm install` this so it's blocking the app from running. So for now in order to see the app I commented this out.

robinv85 replied:
what error do you get? Might be problem with package-lock, or the npm module itself?

anniegiang commented on client/src/pages/CreateUserProfile.js:
```diff
@@ -0,0 +1,219 @@
+        <ProfileFormGroup>
+            <CustomH1 color={theme.colors.darkerGray} fontweight="bold">Create your Profile</CustomH1>
+            <InputGroup>
+                <TextInput label="E-mail" />
```
- I'm not sure if it's a mistake by the design team but only the email field is a gray while the other inputs are royal blue.
- The underline moves surrounding elements when it's active/hovered/focused
- Need the icon that's next to the Address label

figma:
![Screen Shot 2020-05-07 at 2 49 49 PM](https://user-images.githubusercontent.com/42228386/81348118-5847e680-9072-11ea-8985-974e24bd1d19.png)

T-Allen replied:
Yeah, on the Desktop MVP in Figma, there appears to be a screen where the email field is grayed out, and another screen where it is not while the location drop-down menu is open. I am not sure if that is intentional. 

![image](https://user-images.githubusercontent.com/11316853/81350682-546a9300-9077-11ea-8fd5-50f1e33d1048.png)

T-Allen replied:
Actually on this screen, the email field is removed entirely and the email address is displayed in a welcome message as a subheading. I'm assuming this is the intended behavior after a user enters their email address.

T-Allen commented:
> Thanks for working on this! I left some comments ready for your review.
> 
> But before you start, is this PR a duplicate of PR #100 by @CeesWang? I believe that PR was only the mobile version, while this PR is both mobile and desktop. Can you guys clarify with each other?

Kinda, I had mistakenly thought there was no mobile page, so I updated the title. I changed it back to reflect that this is only for desktop. I will probably add the desktop styling to @CeesWang's mobile page instead of creating a new page in my revisions.

anniegiang commented:
@T-Allen Awesome, sounds good. Thanks for clarifying.

anniegiang commented:
> lgtm, does this actually add the page too? Would be nice to get this merged asap so we can continue with the whole user onboarding flow and easily create users and login to test all the other features

If we're in a rush we could merge this since the changes are minor, and then continue to make the changes after it's merged.

T-Allen commented:
> lgtm, does this actually add the page too? Would be nice to get this merged asap so we can continue with the whole user onboarding flow and easily create users and login to test all the other features

@robinv85, unfortunately not. I had only written the styled components and layout for the page with the profile creation form. I also am not sure if this page and the confirmation screens belong in the same file, and I had not added those.

robinv85 commented:
@T-Allen what's the status of this PR?
For reference, probably best if we use the `react-hook-form` library and avoid redux but use `useReducer` hooks as much as possible, see my recently opened PR that refactors the login & signup page 415
For now I've only kept the logged-in user in redux with "isauthenticated" and "emailVerified" fields, we will also have to add the check if a user in our db exists to handle the redirection to this page when needed. I would like to help out with this...

T-Allen commented:
@robinv85 I've addressed each of the style changes except for the the exact shape of the check icon in the check-boxes which does not seem to be a priority right now. Also, I had to target some of the parent elements to hide the toolbar on this screen, as was shown in the designs.  
![image](https://user-images.githubusercontent.com/11316853/82592532-ec777a80-9b55-11ea-85b4-406e3c81ed5e.png)

robinv85 commented:
@T-Allen ok cool, can we get this merged asap so we can continue to finish the whole onboarding flow and the rest of the functionalities for logged-in users? because i think we might be missing some more tickets there for things that belong in MVP

T-Allen commented:
@robinv85 For sure. Unfortunately, this PR is linked to my forked repo which does not have appropriate permissions. I'm going to close this, and resolve the issue with a new PR.
